### PR TITLE
post process all runs calling service over a list of run names

### PR DESCRIPTION
**Pull request by diitaz93**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/cli/post_process/utils.py [hunk 1]
```diff
@@ -21,3 +21,11 @@ def get_post_processing_service_from_run_name(
             f"Run name {run_name} does not match with any known sequencing run name pattern"
         ) from error
     return getattr(context.post_processing_services, device)
+
+
+def get_all_run_names(context: CGConfig) -> list[str]:
+    """Get all run names from the sequencing directories."""
+    run_names: list[str] = []
+    for run_names_service in context.run_names_services.model_dump().values():
```
Comment (islean) [inline cg/cli/post_process/utils.py:29]: This line seems to indicate that we perhaps do not want a basemodel containing all these services? Seems easier to just have them in a list.

File: cg/services/run_devices/pacbio/post_processing_service.py [hunk 2]
```diff
@@ -49,11 +51,19 @@ def __init__(
         to_raise=PostProcessingError,
     )
     def post_process(self, run_name: str, dry_run: bool = False) -> None:
-        LOG.info(f"Starting PacBio post-processing for run: {run_name}")
+        LOG.info(f"Starting Pacbio post-processing for run: {run_name}")
         run_data: PacBioRunData = self.run_data_generator.get_run_data(
             run_name=run_name, sequencing_dir=self.sequencing_dir
         )
+        if self._is_run_post_processed(run_name):
```
Comment (islean) [inline cg/services/run_devices/pacbio/post_processing_service.py:58]: I would not have this check here, I would have it where we fetch the runs to post process.

File: cg/services/run_devices/run_names/pacbio.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    def get_run_names(self) -> list[str]:
+        """
+        Get all the run names from the PacBio sequencing directory in the form
+        <sequencing_run>/<SMRTcell>, for example:
+            r84202_20240913_121403/1_C01
+        """
+
+        run_names = []
+        for run_folder in Path(self.run_directory).iterdir():
+            if run_folder.is_dir():
```
Comment (islean) [inline cg/services/run_devices/run_names/pacbio.py:17]: I do think that this check indicates that we have a trust issue which can cause harm. I have added a discussion point to the coding guidelines. I would let the code break and ensure that the structure is correct if this check fails

File: cg/cli/post_process/post_process.py [hunk 4]
```diff
@@ -35,4 +35,25 @@ def post_process_sequencing_run(context: CGConfig, run_name: str, dry_run: bool)
...
+@DRY_RUN
+@click.pass_obj
+def post_process_all_runs(context: CGConfig, dry_run: bool) -> None:
+    """Post-process all runs from the instruments."""
+    exit_success: bool = True
+    run_names: list[str] = get_all_run_names(context=context)
+    for run_name in run_names:
+        try:
+            post_processing_service: PostProcessingService = (
+                get_post_processing_service_from_run_name(context=context, run_name=run_name)
```
Comment (islean) [inline cg/cli/post_process/post_process.py:48]: This also feels backwards to me. I would prefer if we in this approach already kept the run names grouped per post processing service as in the board sketch

**Review by islean — COMMENTED** [review]
Some initial comments

File: cg/services/run_devices/pacbio/post_processing_service.py (hunk 2 above)
Comment (diitaz93) [reply cg/services/run_devices/pacbio/post_processing_service.py:58]: Here I disagree, as the purpose of the name service is to use it in other services, it is not exclusively a step in the post-processing

**Comment by diitaz93** [comment]
Closing in favour of https://github.com/Clinical-Genomics/cg/pull/3845

**Pull request closed without merging**